### ISSUE TITLE
Allow `print` stdout and stderr to be redirected

### ISSUE THREAD

**Issue by NotTheDr01ds** [nushell/nushell#12686]
### Related problem

Current only *external* commands' stdout/stderr can be redirected.

```Nushell
print -e "Oops" e> error.log
```

... does not work.  Output is still on the terminal.  See "Additional Context" below for details on why this is needed.

### Describe the solution you'd like

`print`/`print -e` output should be redirectable via `o>`/`e>`/etc. allowing the above example to work.

Side-note: `print` is the *only* command in Nushell whose explicit purpose is to output to stdout/stdout.

Side-note:  If this is implemented, then `tee` also needs to be reviewed/updated to make sure it can handle `print`/`print -e` as it has the same issue.

### Describe alternatives you've considered

Of course, I could wrap each `print` statement in a subshell and force it to be an external command from the *current* shell's perspective:

```Nushell
nu -c 'print -e Error Text'
```

That's a bit cringeworthy, IMHO.

I also experimented with `save` since it has an option for capturing stderr to a separate file, but:

1. It doesn't work for `print -e` either
2. Requires some Nushell structured data in order to pass to a `save` file in the first place, even before you could consider redirecting `stdout`

### Additional context and details

A common pattern for a Nushell custom command is:

* When successful, a status message is `print`ed to stdout
* When there's an error condition or something unexpected, `print -e` debugging information to stderr

For example, I have something like the following logic in a custom command at the moment:

```Nushell
def getLatestUpdatesThroughAPI [] {
  try {
    let res = http get ...
  } catch {|e|
    print -e $'Error: ...'
  }

  if $res.status? == 200 {
    print $'Retrieved ($res.body | length) records'
    $res.body | into sqlite ...
  } else {
    print -e $res
  }
}

loop {
  getLastestUpdatesThroughAPI
  sleep 10s
}
```

Error conditions are rare, so my Tmux buffer contains about 2 hours of (usually) normal updates.

So to make sure I see any error conditions, I'd like to do the following:

```Nushell
loop {
  getLatestUpdatesThroughAPI
  sleep 10s
} e> api_errors.log
```

However, that doesn't work, since only the stderr from *external* commands is redirected.  `print -e` is still output to the terminal.

**Comment by NotTheDr01ds** [comment]
Of course, *other* Nushell commands can output to stderr.  I just noticed in my history the (expected):

> `Warning: The first row contains a null value, which has an unknown SQL type. Null values will be assumed to be TEXT columns. If this is undesirable, you can create the table first with your desired schema.`

This should probably be redirectable via `e>` as well, but I doubt it is currently.

**Comment by sholderbach** [comment]
With the back-and-forth iterations of what the semantics of our `echo` are  (print/pure value forward) and when you need `print` or don't need them, and the vague ideas how you could deal with running a script with `nu script.nu | what-is-processed-here`, I think it is worth laying out the full semantics of how we incorporate the traditional POSIX STDOUT/STDERR semantics with the Nushell `PipelineData` in a document.

Strongly advising against implementing something specific to close this issue before we have a blessed design.

**Comment by NotTheDr01ds** [comment]
@sholderbach I understand your reasoning there, but is redirection of internal Nu commands really tied to the decision of *what* is doing the "sending" to stdout/stderr?  This *seems* to me to be something that is going to need to be solved in the redirection mechanism rather than the command-level.  Is that not the case? (Honest, non-rhetorical question).

That said, this isn't something *critical* since there are hacky workarounds, so it *can* certainly wait.  I just know that `echo` discussions have gone round-and-round for a while now ;-)

**Comment by sholderbach** [comment]
> This _seems_ to me to be something that is going to need to be solved in the redirection mechanism rather than the command-level.

Yes, absolutely this should be solved as a consistent semantic instead of piecemeal by command solutions (e.g. just changing something for `print`)

But currently our engine warning/error messages get emitted by the nushell process through the usual means for a process i.e. simply writing to the STDERR. Some of those may be worth considering to allow a redirection, others may only make sense to show the user/the service running `nu`

The primary concern I have around all those rounds we had so far is that we should make sure that we have a consistent and if possible simple enough story around *the* nushell pipeline and the three special POSIX file descriptors `STDIN`, `STDOUT`, `STDERR`. It should be approachable both coming from a prototypical POSIX shell like bash, writing your own CLI programs with those streams, and for folks just starting out on the shell with Nushell.
